### PR TITLE
media-sound/flacon: verbump to 3.1.1

### PR DESCRIPTION
**Pull request by Coacher**
Fixes incorrect offset for long disks due to integer overflow.

Package-Manager: Portage-2.3.6, Repoman-2.3.3

### CODE REVIEW

File: media-sound/flacon/flacon-3.1.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+# Ignore rudimentary et, uz@Latn, zh_TW translation(s).
+PLOCALES="cs cs_CZ de es es_MX fr gl hu it ja_JP lt nb nl pl pl_PL pt_BR pt_PT ro_RO ru sr sr@latin tr uk zh_CN"
+# Tests require lots of disk space
+CHECKREQS_DISK_BUILD=10G
```
Comment (mgorny) [inline media-sound/flacon/flacon-3.1.1.ebuild]: Move this closer to `pkg_*`, so people can see instantly that it applies only USE-conditionally. Also, it's not needed before `inherit`.

File: media-sound/flacon/flacon-3.1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	local mycmakeargs=(
+		-DUSE_QT4="$(usex !qt5)"
+		-DUSE_QT5="$(usex qt5)"
+		-DBUILD_TESTS="$(usex test)"
+	)
+	cmake-utils_src_configure
+}
+
+src_test() {
+	virtx "${BUILD_DIR}/tests/${PN}_test"
```
Comment (mgorny) [inline media-sound/flacon/flacon-3.1.1.ebuild]: Don't you usually want to fail when the tests fail?

File: media-sound/flacon/flacon-3.1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+pkg_preinst() {
+	gnome2_icon_savelist
```
Comment (mgorny) [inline media-sound/flacon/flacon-3.1.1.ebuild]: No longer necessary.

File: media-sound/flacon/flacon-3.1.1.ebuild (hunk 1 above)
Comment (Coacher) [inline media-sound/flacon/flacon-3.1.1.ebuild]: I'd like to keep variables that control eclasses' behaviour close to inherits.

Comment (Coacher) [inline media-sound/flacon/flacon-3.1.1.ebuild]: Also there's a perfectly good comment explaining things.

File: media-sound/flacon/flacon-3.1.1.ebuild (hunk 2 above)
Comment (Coacher) [inline media-sound/flacon/flacon-3.1.1.ebuild]: virtx already dies when tests fail, see https://github.com/gentoo/gentoo/blob/63b98662a14346b785c7e96c61d644a971779a70/eclass/virtualx.eclass#L219

File: media-sound/flacon/flacon-3.1.1.ebuild (hunk 3 above)
Comment (Coacher) [inline media-sound/flacon/flacon-3.1.1.ebuild]: Fixed.

**Comment by Coacher** [comment]
Thank you, @mgorny.

**Pull request closed without merging**